packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -257,8 +257,12 @@ export const useGridVirtualScroller = () => {
       // otherwise we would call an update directly on mount
       const isReady = gridDimensionsSelector(apiRef).isReady;
       if (isReady && didRowsIntervalChange) {
+        const isFirstRenderContextHydration = previousRowContext.current === EMPTY_RENDER_CONTEXT;
         previousRowContext.current = nextRenderContext;
-        apiRef.current.publishEvent('renderedRowsIntervalChange', nextRenderContext);
+
+        if (!isFirstRenderContextHydration) {
+          apiRef.current.publishEvent('renderedRowsIntervalChange', nextRenderContext);
+        }
```

Comment:
Should this be ported for the new virtualizer code? Also note that this could be considered a breaking change, that event is public and its behavior is changing here.